processing/src/main/java/org/apache/druid/query/aggregation/mean/DoubleMeanBufferAggregator.java

```diff
@@ -51,6 +52,9 @@ public void aggregate(ByteBuffer buf, int position)
   {
     Object update = selector.getObject();
 
+    if (update == null && NullHandling.replaceWithDefault() == false) {
```

Comment:
Somehow setting 
```
boolean status = selector.isNull();
    if (status == true && NullHandling.replaceWithDefault() == false) {
      return;
    }
```
was not working but setting the update as null works correctly

Reply:
yeah, this agg is a bit strange in that it supports both numeric primitives and the complex merging; isNull only works for numeric selectors, other selector types need to check that the value itself is null.